platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/migratable/ApplicationProcessor.java

```diff
@@ -0,0 +1,220 @@
+  public void processMemoryApplications(Map<String, JsonApplication> japps, TaskList tasks) {
+    LOGGER.debug("Processing applications defined in memory");
+    // check all apps in memory and either start or stop them based on the exported info
+    for (final Application app : service.getApplications()) {
+      final String name = app.getName();
+      final JsonApplication japp = japps.remove(name); // remove from japps when we find it
+
+      processApplication(japp, app, tasks);
+    }
+  }
+
+  /**
+   * Processes applications that were left over after having dealt with what is in memory. The
+   * implementation will try to find the missing applications (although unlikely to be found). If
+   * found, they will be processed by recording tasks to start them if they were originally started.
+   *
+   * @param report the report where to record errors
+   * @param japps the set of apps on the original system that were not yet processed from memory
+   * @param tasks the task list where to record tasks to be executed
+   * @return <code>false</code> if we failed to retrieve from memory a left over application; <code>
+   *     true</code> if everything was ok
+   */
+  public boolean processLeftoverExportedApplications(
+      ProfileMigrationReport report, Map<String, JsonApplication> japps, TaskList tasks) {
+    LOGGER.debug("Processing leftover exported applications");
+    boolean processed = true; // until proven otherwise
+
+    // check if there are anything left from the exported info that should be started or that is
+    // not installed
+    for (final JsonApplication japp : japps.values()) {
+      final String name = japp.getName();
+
+      if (japp.isStarted()) {
+        final Application app = getApplication(report, name);
+
+        if (app == null) { // app not available!!!
+          // error already recorded from getApplication()
+          processed = false;
+        } else {
+          processApplication(japp, app, tasks);
+        }
+      } else { // we shall let the features and bundles handle the low-level stuff
+        LOGGER.debug("Skipping application '{}'; not installed", name);
+      }
+    }
+    return processed;
+  }
+
+  /**
+   * Processes the specified application by comparing its state in memory to the one from the
+   * original system and determining if it needs to be started.
+   *
+   * @param japp the original application information or <code>null</code> if it was not installed
+   * @param app the current application from memory
+   * @param tasks the task list where to record tasks to be executed
+   */
+  public void processApplication(@Nullable JsonApplication japp, Application app, TaskList tasks) {
+    final String name = app.getName();
+
+    if (japp != null) {
+      if (japp.isStarted()) {
+        final boolean started = service.isApplicationStarted(app);
+
+        if (!processStartedApplication(app, name, started, tasks)) {
```

Comment:
❓ Should we rename `processStartedApplication` to something like `startApplication`, call it only if the application wasn't started and get rid of the `started` flag altogether?
That would simplify the logic a little, eliminate the `boolean` parameter and the returned `boolean` as well.

Reply:
This was done to be consistent with the other processor classes. In any case, this class is being removed in the next PR.